### PR TITLE
Update css-flexbox-tests.txt for new/deleted tests

### PR DESCRIPTION
**Pull request by davidsgrogan**
The new tests seem reasonable to include, with two hesitations:

 * table-as-item* -- We explicitly decided to exclude tables as a focus area from compat2021. Making Blink pass these `flexbox/table-as-item` tests required a bunch of work in the tables code. Maybe we shouldn't penalize Webkit for an area that we agreed to deprioritize? Maybe we should exclude any of these table tests, including preexisting, that any engine fails?
 * flexbox-min-width-auto-00{5,6} -- Currently all engines pass, but Blink thinks the tests are wrong and plans to change them as long as the behavior change is web compatible (see https://github.com/web-platform-tests/wpt/issues/28805)

@dholbert, thoughts on the updated test list? Any we should remove from the original list because they are invalid?

### CODE REVIEW

**Comment by davidsgrogan** [comment]
@stephenmcgruer, can you review after we hear from @dholbert and/or @aethanyc?

**Comment by dholbert** [comment]
Just looking at the new tests that fail in Firefox (to sniff out ones that seem like they might be invalid), the list looks reasonable I think.

These are the failures I found & looked at:
 - the `aspect-ratio-intrinsic-size-*` tests look valid & seem to be pointing out a real Firefox bug.
 - the test `table-as-item-min-height-1.html` also looks like a real & subtle Firefox bug. (It looks like `min-width` is influencing our resolved flex-basis for the table, I think.)
 - the test `table-as-item-stretch-cross-size-5.html` also looks like a real Firefox bug (not entirely sure what's going on there; we seem to be disregarding the flex-basis that's been set on the table flex item, it seems)
 
Having said that: I do agree with your hesitation about including many tables-as-flex-items tests as part of the compat2021 test suite here; there are still some rough edges there & there are some authors expecting legacy behavior as shown by e.g. https://bugzilla.mozilla.org/show_bug.cgi?id=1701447 (and other resolved-as-invalid regressions from https://bugzilla.mozilla.org/show_bug.cgi?id=1674302 ).

It's of course nice to have interop on this, but it's not as clear (compared to other spec areas) that rough edges & webcompat demands are fully hammered out here.

**Comment by davidsgrogan** [comment]
Ok, since we share the same hesitation about the table tests, I left only the ones where the engines already behave the same.

I also commented out flexbox-min-width-auto-00{5,6} for the reasons mentioned in my previous comment.

**Comment by aethanyc** [comment]
I defer to @dholbert's judgement on which tests to include in the list.

**Comment by davidsgrogan** [comment]
I think we're ready to merge this now. @foolip or @stephenmcgruer (whomever we should be pestering about this), ready when you are.

**Review by foolip — APPROVED** [review]

**Comment by stephenmcgruer** [comment]
@foolip - reminder, you said you wanted to regen the results ;)

**Comment by foolip** [comment]
@stephenmcgruer I seem to have forgotten what I wanted 😊

I guess that when https://github.com/Ecosystem-Infra/wpt-results-analysis/blob/main/.github/workflows/update_gh_pages.yml runs next this change will take effect, by regen do you mean recomputing older data points?

**Comment by stephenmcgruer** [comment]
I assume you wanted to regen *right away*? Because yeah otherwise we still recompute all data (because it's still pretty cheap - so far)

**Comment by foolip** [comment]
Aha, so I looked at recent action runs and thought it looked pretty regular, but looking closer it's only 3 times a week. I've sent https://github.com/Ecosystem-Infra/wpt-results-analysis/pull/48 to make it easy to recompute directly after merging a change like this.

**Comment by foolip** [comment]
With https://github.com/Ecosystem-Infra/wpt-results-analysis/pull/48 landed, @stephenmcgruer triggered the update manually, and https://github.com/Ecosystem-Infra/wpt-results-analysis/commit/2bfe3b7b378b08dcf5f1b0978f7dfe4daf431fed looks to have recomputed all the Flexbox numbers as expected.

I'm not sure what the effect on the shape of https://wpt.fyi/compat2021?feature=css-flexbox was, but looking at `data/compat2021/css-flexbox-experimental.csv` it looks like:
- Chrome 91.06% &rarr; 91.39%
- Firefox 98.06% &rarr; 98.11%
- Safari 87.17% &rarr; 86.85%

If this isn't in line with the expected effect of including these tests, please shout!

**Comment by davidsgrogan** [comment]
> I'm not sure what the effect on the shape of https://wpt.fyi/compat2021?feature=css-flexbox was, but looking at `data/compat2021/css-flexbox-experimental.csv` it looks like:
> 
> * Chrome 91.06% → 91.39%
> * Firefox 98.06% → 98.11%
> * Safari 87.17% → 86.85%

@foolip, what is the denominator for these percentages? Does it count each subtest in a file separately, or just the number of files? Like does [css/css-flexbox/abspos/flex-abspos-staticpos-justify-content-007.html](https://wpt.fyi/results/css/css-flexbox/abspos/flex-abspos-staticpos-justify-content-007.html?label=experimental&label=master&aligned) contribute 1 to the denominator or 24?

**Comment by davidsgrogan** [comment]
There seem to be 1083 non-commented lines in https://github.com/Ecosystem-Infra/wpt-results-analysis/blob/main/compat-2021/css-flexbox-tests.txt but I can't find whole numbers that map to the percentages you pasted from the .csv file.

E.g. for Firefox:
1062 / 1083 = 0.9806
1063 / 1083 = 0.9815

**Comment by foolip** [comment]
@davidsgrogan I didn't compute any fraction but rather read out the computed number from `data/compat2021/css-flexbox-experimental.csv`.

I don't have access to my computer and can't do meaningful digging now, but do the numbers seem off? They should be the same ones that show up in the wpt.fyi graphs, if I'm not mistaken.

**Comment by stephenmcgruer** [comment]
> @foolip, what is the denominator for these percentages? Does it count each subtest in a file separately, or just the number of files? E.g., does css/css-flexbox/abspos/flex-abspos-staticpos-justify-content-007.html contribute 1 to the denominator or 24?

Each file can contribute at most 1 'point' to the total score for a component. Subtests are handled by splitting that '1' point equally across subtests, and giving it a fraction passed on passing subtests (e.g. passing `17 / 25` will contribute 0.68 to the score rather than 1). [Code link](https://github.com/Ecosystem-Infra/wpt-results-analysis/blob/main/compat-2021/main.js#L150) (which really needs a few lines of documentation on the method...)

The denominator is the total number of test files listed in the `.txt` file - **which may be more test files than are found in a given run** (as tests are added or deleted). This is discussed in the [code here](https://github.com/Ecosystem-Infra/wpt-results-analysis/blob/main/compat-2021/main.js#L187) and is a non-trivial trade-off.


As with all metrics, incidentally, the compat2021 metric is **so** far from perfect, because every decision biases one in some direction :). Thus I tend to encourage folks not to worry about small differences, and focus more on things moving in a positive direction and the 'big picture' (e.g. a difference of say 10% between two browsers is _probably_ significant, whilst 1% could easily be noise).

**Comment by davidsgrogan** [comment]
> @davidsgrogan I didn't compute any fraction but rather read out the computed number from `data/compat2021/css-flexbox-experimental.csv`.

Yeah, my question was about how that number was originally computed.

> Each file can contribute at most 1 'point' to the total score for a component. Subtests are handled by splitting that '1' point equally across subtests, and giving it a fraction passed on passing subtests (e.g. passing `17 / 25` will contribute 0.68 to the score rather than 1). [Code link](https://github.com/Ecosystem-Infra/wpt-results-analysis/blob/main/compat-2021/main.js#L150) (which really needs a few lines of documentation on the method...)

Ah, makes sense, I hadn't considered this method. Good explanation, thanks.

> The denominator is the total number of test files listed in the `.txt` file - **which may be more test files than are found in a given run** (as tests are added or deleted). This is discussed in the [code here](https://github.com/Ecosystem-Infra/wpt-results-analysis/blob/main/compat-2021/main.js#L187) and is a non-trivial trade-off.
> 
> As with all metrics, incidentally, the compat2021 metric is **so** far from perfect, because every decision biases one in some direction :). Thus I tend to encourage folks not to worry about small differences, and focus more on things moving in a positive direction and the 'big picture' (e.g. a difference of say 10% between two browsers is _probably_ significant, whilst 1% could easily be noise).

That explanation for why not to worry about small differences totally makes sense. At the same time, for OKRs, I've been asked to estimate the effect of my Q3 Blink work on this metric, which will only be a few percentage points. So 🤷, I'll do my best at estimating the small difference.

In related news, I recently landed a patch that I expect will move css-flex a few percentage points from 91 to maybe 94.

**Comment by foolip** [comment]
> I've been asked to estimate the effect of my Q3 Blink work on this metric

It's a good point that this is harder to estimate with the scoring we use. On the flip size, numbers would be more prone to large changes if we didn't have this normalization per test, so I think overall it still makes sene.

> In related news, I recently landed a patch that I expect will move css-flex a few percentage points from 91 to maybe 94.

Awesome! What kinds of bugs are addressed by that? Something to let web developers know about?

**Comment by davidsgrogan** [comment]
> > I've been asked to estimate the effect of my Q3 Blink work on this metric
> 
> It's a good point that this is harder to estimate with the scoring we use. On the flip size, numbers would be more prone to large changes if we didn't have this normalization per test, so I think overall it still makes sene.

Oh totally agree that this is a good calculation method. I was more referring to the fact that small percentage differences aren't really meaningful but yet my OKRs include a small percentage difference :)

> > In related news, I recently landed a patch that I expect will move css-flex a few percentage points from 91 to maybe 94.
> 
> Awesome! What kinds of bugs are addressed by that? Something to let web developers know about?

It was actually [one](https://crbug.com/1011718) you ran into last year -- adding support for css-align-3 positional alignment keywords (start, end, self-start, left, etc). Joe Medley knows that it's shipping in M93 so I assume he is going to put it in the release notes.

**Pull request merged**
